app/src/main/java/com/nervousfish/nervousfish/service_locator/ServiceLocatorNoNetwork.java

```diff
@@ -0,0 +1,119 @@
+package com.nervousfish.nervousfish.service_locator;
+
+import com.nervousfish.nervousfish.ConstantKeywords;
+import com.nervousfish.nervousfish.modules.constants.Constants;
+import com.nervousfish.nervousfish.modules.constants.IConstants;
+import com.nervousfish.nervousfish.modules.cryptography.EncryptorAdapter;
+import com.nervousfish.nervousfish.modules.cryptography.IEncryptor;
+import com.nervousfish.nervousfish.modules.cryptography.IKeyGenerator;
+import com.nervousfish.nervousfish.modules.cryptography.KeyGeneratorAdapter;
+import com.nervousfish.nervousfish.modules.database.GsonDatabaseAdapter;
+import com.nervousfish.nervousfish.modules.database.IDatabase;
+import com.nervousfish.nervousfish.modules.filesystem.AndroidFileSystemAdapter;
+import com.nervousfish.nervousfish.modules.filesystem.IFileSystem;
+import com.nervousfish.nervousfish.modules.pairing.AndroidBluetoothHandler;
+import com.nervousfish.nervousfish.modules.pairing.DummyBluetoothHandler;
+import com.nervousfish.nervousfish.modules.pairing.DummyNFCHandler;
+import com.nervousfish.nervousfish.modules.pairing.DummyQRHandler;
+import com.nervousfish.nervousfish.modules.pairing.IBluetoothHandler;
+import com.nervousfish.nervousfish.modules.pairing.INfcHandler;
+import com.nervousfish.nervousfish.modules.pairing.IQRHandler;
+
+import org.greenrobot.eventbus.EventBus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InvalidObjectException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+
+/**
+ * Manages all modules and provides access to them.
+ * It disables Bluetooth and NFC because that cannot be used on an emulator
+ */
+public class ServiceLocatorNoNetwork extends ServiceLocator implements IServiceLocator {
```

Comment:
What do we need a `ServiceLocatorNoNetwork` for? We don't rely on a network...

Also, test code shouldn't affect production code so I think we should not extend `ServiceLocator` here so that the `ServiceLocator` class can be final again 😄